docs/source/sparse.rst

```diff
@@ -1,146 +1,382 @@
-    .. method:: spmm
-    .. method:: sspaddmm
-    .. method:: sspmm
-    .. method:: sub
-    .. method:: sub_
-    .. method:: t_
-    .. method:: to_dense
-    .. method:: transpose
-    .. method:: transpose_
-    .. method:: zero_
-    .. method:: coalesce
-    .. method:: is_coalesced
-    .. method:: _indices
-    .. method:: _values
-    .. method:: _nnz
-
-Functions
-----------------------------------
+   :func:`torch.spmm` and :func:`torch.dsmm` are aliases to :func:`torch.mm`.
+
+.. note::
+
+   Currently, Torch does not support matrix multiplication with the
+   layout signature ``M[strided] @ M[sparse_coo]``. However,
+   applications can still compute this using the matrix relation ``D @
+   S == (S.t() @ D.t()).t()``.
+
+.. class:: Tensor()
+   :noindex:
 
+   The following methods are specific to :ref:`sparse tensors <sparse-docs>`:
+
+    .. autoattribute:: is_sparse
+    .. automethod:: dense_dim
+    .. automethod:: sparse_dim
+    .. automethod:: sparse_mask
+    .. automethod:: sparse_resize_
+    .. automethod:: sparse_resize_and_clear_
+    .. automethod:: to_dense
+    .. automethod:: to_sparse
+
+   The following methods are specific to :ref:`sparse COO tensors
+   <sparse-coo-docs>`:
+
+    .. automethod:: coalesce
+    .. automethod:: is_coalesced
+    .. automethod:: indices
+    .. automethod:: values
+
+The following :class:`torch.Tensor` methods support :ref:`sparse COO
+tensors <sparse-coo-docs>`:
+
+:meth:`~torch.Tensor.add`
+:meth:`~torch.Tensor.add_`
+:meth:`~torch.Tensor.addmm`
+:meth:`~torch.Tensor.addmm_`
+:meth:`~torch.Tensor.any`
+:meth:`~torch.Tensor.asin`
+:meth:`~torch.Tensor.asin_`
+:meth:`~torch.Tensor.arcsin`
+:meth:`~torch.Tensor.arcsin_`
+:meth:`~torch.Tensor.bmm`
+:meth:`~torch.Tensor.clone`
+:meth:`~torch.Tensor.deg2rad`
+:meth:`~torch.Tensor.deg2rad_`
+:meth:`~torch.Tensor.detach`
+:meth:`~torch.Tensor.detach_`
+:meth:`~torch.Tensor.dim`
+:meth:`~torch.Tensor.div`
+:meth:`~torch.Tensor.div_`
+:meth:`~torch.Tensor.floor_divide`
+:meth:`~torch.Tensor.floor_divide_`
+:meth:`~torch.Tensor.get_device`
+:meth:`~torch.Tensor.index_select`
+:meth:`~torch.Tensor.isnan`
+:meth:`~torch.Tensor.log1p`
+:meth:`~torch.Tensor.log1p_`
+:meth:`~torch.Tensor.mm`
+:meth:`~torch.Tensor.mul`
+:meth:`~torch.Tensor.mul_`
+:meth:`~torch.Tensor.mv`
+:meth:`~torch.Tensor.narrow_copy`
+:meth:`~torch.Tensor.neg`
+:meth:`~torch.Tensor.neg_`
+:meth:`~torch.Tensor.negative`
+:meth:`~torch.Tensor.negative_`
+:meth:`~torch.Tensor.numel`
+:meth:`~torch.Tensor.rad2deg`
+:meth:`~torch.Tensor.rad2deg_`
+:meth:`~torch.Tensor.resize_as_`
+:meth:`~torch.Tensor.size`
+:meth:`~torch.Tensor.pow`
+:meth:`~torch.Tensor.square`
+:meth:`~torch.Tensor.smm`
+:meth:`~torch.Tensor.sspaddmm`
+:meth:`~torch.Tensor.sub`
+:meth:`~torch.Tensor.sub_`
+:meth:`~torch.Tensor.t`
+:meth:`~torch.Tensor.t_`
+:meth:`~torch.Tensor.transpose`
+:meth:`~torch.Tensor.transpose_`
+:meth:`~torch.Tensor.zero_`
+
+
+Sparse tensor functions
++++++++++++++++++++++++
```

Comment:
As mentioned previously, what puts a function in the sparse namespace vs the torch namespace?

Reply:
This is not very clear to me and sometimes it feels like the reason has been rather arbitrary.

That said, here are some current criteria for putting a function to `torch.sparse` rather than to `torch` namespace:
- the function has different mathematical behavior on a sparse input compared to the same function on a dense input. Example: `torch.sparse.softmax`
- the function has many different implementations and variable autograd support. Examples: `torch.sparse.addmm`, `torch.sparse.mm`.
- the function is specific to sparse tensors only. No examples.

From the user perspective, `torch.sparse` is not currently very useful (other than using some functions from it).

Reply:
Thank you for the explanation.